packages/core/helper-plugin/lib/src/hooks/useFieldHint/index.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+
+/**
+ * @description
+ * A hook for generating the hint for a field
+ * @param {Object} description - the description of the field
+ * @param {Number} minimum - the minimum length or value of the field
+ * @param {Number} maximum - the maximum length or value of the field
+ * @param {Boolean} isNumber - whether this is a number field
+ */
+const useFieldHint = ({ description, minimum, maximum, isNumber = false }) => {
+  const { formatMessage } = useIntl();
+
+  const [fieldHint, setFieldHint] = useState([]);
+
+  /**
+   * @returns {String}
+   */
+  const buildDescription = () =>
+    description
+      ? formatMessage(
+          { id: description.id, defaultMessage: description.defaultMessage },
+          { ...description.values }
+        )
+      : '';
+
+  /**
+   * Constructs a suitable description of a field's minimum and maximum limits
+   * @returns {Array}
+   */
+  const buildMinMaxHint = () => {
+    const minIsNumber = typeof minimum === 'number';
+    const maxIsNumber = typeof maximum === 'number';
+
+    if (!minIsNumber && !maxIsNumber) {
+      return [];
+    }
+
+    const minMaxDescription = [];
+
+    if (minIsNumber && minimum > 0) {
+      minMaxDescription.push(`min. {minimum}`);
+    }
```

Comment:
Do you think we should print the minimum even if a user has entered 0? I think it might be confusing otherwise.